crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,503 @@
+    /// is the annualized position duration, the rate derivative is:
+    ///
+    /// $$
+    /// r'(x) = \frac{(-p'(x) p(x) t - (1 - p(x)) (p'(x) t))}{(p(x) t)^2} //
+    /// r'(x) = \frac{-p'(x)}{t p(x)^2}
+    /// $$
+    ///
+    /// We return $-r'(x)$ because negative numbers cannot be represented by FixedPoint.
+    fn rate_after_long_derivative_negation(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: FixedPoint,
+    ) -> Result<FixedPoint> {
+        let price = self.calculate_spot_price_after_long(base_amount, Some(bond_amount))?;
+        let price_derivative = self.price_after_long_derivative(base_amount, bond_amount)?;
+        // The actual equation we want to represent is:
+        // r' = -p' / (t p^2)
+        // We can do a trick to return a positive-only version and
+        // indicate that it should be negative in the fn name.
+        Ok(price_derivative / (self.annualized_position_duration() * price.pow(fixed!(2e18))))
+    }
+
+    /// The derivative of the price after a long.
+    ///
+    /// The price after a long that moves shares by $\Delta z$ and bonds by $\Delta y$
+    /// is equal to $P(\Delta z) = \frac{\mu (z_e + \Delta z)}{y - \Delta y}^T$,
+    /// where $T$ is the time stretch constant and $z_e$ is the initial effective share reserves.
+    /// Equivalently, for some amount of `delta_base`$= x$ provided to open a long,
+    /// we can write:
+    ///
+    /// $$
+    /// p(x) = \frac{\mu (z_e + \frac{x}{c} - g(x) - \zeta)}{y_0 - Y(x)}^{T}
+    /// $$
+    /// where $g(x)$ is the [open_long_governance_fee](long::fees::open_long_governance_fee),
+    /// $Y(x)$ is the [long_amount](long::open::calculate_open_long), and $\zeta$ is the
+    /// zeta adjustment.
+    ///
+    /// To compute the derivative, we first define some auxiliary variables:
+    /// $$
+    /// a(x) = \mu (z_e + \frac{x}{c} - g(x) - \zeta) \\
+    /// b(x) = y_0 - Y(x) \\
+    /// v(x) = \frac{a(x)}{b(x)}
+    /// $$
+    ///
+    /// and thus $p(x) = v(x)^T$. Given these, we can write out intermediate derivatives:
+    ///
+    /// $$
+    /// a'(x) = \frac{\mu}{c} - g'(x) \\
+    /// b'(x) = -Y'(x) \\
+    /// v'(x) = \frac{b a' - a b'}{b^2}
+    /// $$
+    ///
+    /// And finally, the price after long derivative is:
+    ///
+    /// $$
+    /// p'(x) = v'(x) T v(x)^(T-1)
+    /// $$
+    ///
+    fn price_after_long_derivative(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: FixedPoint,
+    ) -> Result<FixedPoint> {
+        // g'(x)
+        let gov_fee_derivative = self.governance_lp_fee()
+            * self.curve_fee()
+            * (fixed!(1e18) - self.calculate_spot_price());
+
+        // a(x) = u (z_e + x/c - g(x) - zeta)
+        let inner_numerator = self.mu()
+            * (self.ze() + base_amount / self.vault_share_price()
+                - self.open_long_governance_fee(base_amount)
+                - self.zeta().into());
```

Comment:
I could be missing something, but as I said above, I think that this zeta term shouldn't be here since $z_e = z - \zeta$. The tests are probably passing because we don't allow positions to mature prior to using the targeted long. 

We should add some of these tests IMO. You'll want to manually checkpoint to automatically mature these positions since the Rust SDK doesn't handle checkpoints well right now.